test/system_test.py

```diff
@@ -20,9 +20,14 @@ def devirtualize_env():
     # VIRTUAL_ENV and PATH
 
     # Remove 'VIRTUAL_ENV/bin'
-    virtual_env_bin = os.path.join(os.environ['VIRTUAL_ENV'], 'bin')
-    os.environ.pop('VIRTUAL_ENV')
-    os.environ['PATH'] = os.environ['PATH'].replace(virtual_env_bin, '')
+    try:
+        virtual_env_bin = os.path.join(os.environ['VIRTUAL_ENV'], 'bin')
+        os.environ.pop('VIRTUAL_ENV')
+        os.environ['PATH'] = os.environ['PATH'].replace(virtual_env_bin, '')
+
+    # happens if test are run in serial-mode because cannot unset var twice
+    except KeyError:
+        pass
```

Comment:
Ideally we devirtualize the environment while pytest is still serial, i.e. right after parsing args.
Because I think our testing will be more reliable if it is isolated from the env it was called.

Reply:
Or we start a new shell that doesn't inherit the env.